@xen-orchestra/lite/src/libs/xapi.ts

```diff
@@ -202,4 +229,49 @@ export default class XapiConnection extends EventEmitter {
 
     return _xapi.call(method, ...args)
   }
+
+  async createNetworks(
+    newNetworks: [
+      {
+        bondMode?: string
+        MTU?: string
+        name_description: string
+        name_label: string
+        pifsId?: string[]
+        VLAN?: string
+      }
+    ]
+  ): Promise<(string | undefined)[]> {
+    const pifs = this.objectsByType.get('PIF')
+    return Promise.all(
+      newNetworks.map(async ({ bondMode, pifsId, ...newNetwork }) => {
+        let networkRef: string | undefined
+        try {
+          networkRef = (await this.call('network.create', {
+            ...newNetwork,
+            other_config: { automatic: 'false' },
+          })) as string
+          await this.barrier(networkRef)
+          const networkId = this.objectsByType.get('network')?.find(({ $ref }) => $ref === networkRef)?.$id
+
+          if (pifsId === undefined) {
+            return networkId
+          }
+          if (bondMode !== undefined) {
+            await Promise.all(
+              pifsId.map(pifId => this.call('Bond.create', networkRef, pifs?.get(pifId)?.$network.PIFs, '', bondMode))
+            )
+          } else {
+            await this.call('pool.create_VLAN_from_PIF', pifs?.get(pifsId[0])?.$ref, networkRef, newNetwork.VLAN)
+          }
+          return networkId
+        } catch (error) {
+          if (networkRef !== undefined) {
+            await this.call('network.destroy', networkRef)
+          }
```

Comment:
We can start using [`defer`](https://www.npmjs.com/package/golike-defer) since I think we'll need it multiple times.